### DISCUSSION TITLE
Precommit Hook With Husky

### DISCUSSION THREAD

**Discussion by JimmayVV** [epicweb-dev/epic-stack#358] [General]
Hey Everyone,

Thanks, Kent, for providing such a great technical solution here for others to draw inspiration from. I have a quick question, is there a particular reason you do not have a precommit hook strategy in the repo (like with husky)?

Reason I ask is because I just ran into a situation where for some reason my prettier extension in vscode was accidentally disabled and wasn't auto-formatting my code like I thought it was.

I don't want to _assume_ that all apps should have precommit hooks, but I know in your testing javascript course, you demonstrate doing just that, and I happened to like it.

So I figured I'd ask. Has your opinion changed about the utility of precommit hooks? Or is there a workflow you utilize that supersedes that need?

Thanks!
Jimmy

**Comment by kentcdodds** [comment]
Hi @JimmayVV,

I don't know that I would say my opinion has changed on the value of precommit hooks, but I don't think it's worth it all the time and in this app I don't feel like it's worth the extra complexity. I definitely welcome anyone to make [an example](https://github.com/epicweb-dev/epic-stack/blob/main/docs/examples.md) of how to add them though!
